caom2-artifact-resolvers/src/test/java/ca/nrc/cadc/caom2/artifact/resolvers/CadcResolverTest.java

```diff
@@ -0,0 +1,182 @@
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*  $Revision: 5 $
+*
+************************************************************************
+*/
+
+package ca.nrc.cadc.caom2.artifact.resolvers;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.net.Traceable;
+import ca.nrc.cadc.util.Log4jInit;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+/**
+ * @author adriand
+ */
+public class CadcResolverTest {
+    private static final Logger log = Logger.getLogger(CadcResolverTest.class);
+    private static final String SI_URL = "https://unittest.com/global/files";
+
+
+    static {
+        Log4jInit.setLevel("ca.nrc.cadc", Level.INFO);
+    }
+
+    private static final String FILE_URI = "cadc:FOO/bar";
+
+    private final CadcResolver cadcResolver;
+
+    public CadcResolverTest() {
+        cadcResolver = getCadcResolver();
+
+    }
+
+    private CadcResolver getCadcResolver() {
+        // override the capabilities method
+        return new CadcResolver() {
+            @Override
+            public URL getServiceURL(AuthMethod am) throws URISyntaxException {
+                try {
+                    return new URL(SI_URL);
+                } catch (MalformedURLException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+    }
+
+    @Test
+    public void testTraceable() {
+        Assert.assertTrue(cadcResolver instanceof Traceable);
+    }
+    
```

Comment:
Should be a test where setAuthMethod is never called to make sure that does not result in an NPE. That's actually the normal usage (getting AuthMethod from Subject)